### ISSUE TITLE
Missing Control Board interfaces bindings since PR #1801

### ISSUE THREAD

**Issue by nunoguedelha** [robotology/yarp#1812]
**Describe the bug**
When generating the yarp bindings with Yarp on the latest commit https://github.com/robotology/yarp/commit/e4fa62170ec3f0de73030846d4315716a217ff7b, all the bindings related to the Control Board interfaces are missing (IAmplifierControl,IAxisInfo,IControlLimits,etc) as well as the related VOCABs.

**To Reproduce**
Regenerate bindings using through the usual procedure.

**Expected behavior**
The interfaces exist, so the bindings should be there.

**Screenshots**
NA

**Configuration (please complete the following information):**
- OS: Irrelevant
- Yarp commit: https://github.com/robotology/yarp/commit/e4fa62170ec3f0de73030846d4315716a217ff7b
- SWIG commit: https://github.com/robotology-dependencies/swig/commit/7e61db6b5dc841027439081aed3bf8c36c9e1920

Copied people who generated or generate bindings on a regular basis.

CC @randaz81 @traversaro @lrapetti @fjandrad @diegoferigo

**Comment by nunoguedelha** [comment]
Hi @randaz81 , actually [this commit](https://github.com/robotology/yarp/commit/e876d1b) introduced the issue for two reasons:
- the change (interfaces moved to new header files) was not reflected in `yarp/dev/all.h`
- the change was not reflected in `yarp.i` which includes `yarp/dev/all.h` and defines the interfaces to be binded, SWIG doesn't find anymore the definitions of `IAmplifierControl`, `IAxisInfo`, `IControlLimits`, etc.

The fix is quite simple, updating `yarp/dev/all` and adding in `yarp.i`, next to the header inclusion `%include <yarp/dev/ControlBoardInterfaces.h>` the inclusion of the new headers.

Regarding `yarp.i`, if it is updated this way by the same developer changing **Yarp**, or at least if a red flag is raised, it would saves us a lot of time.

@traversaro maybe we should auto-generate the block of instructions `%include <xxxxxxx.h>` from the files `<yarp/os/all.h>`, `<yarp/sig/all.h>`, `<yarp/dev/all.h>`, this way the update of `yarp/xxx/all.h` would be sufficient.

**Comment by traversaro** [comment]
> @traversaro maybe we should auto-generate the block of instructions `%include <xxxxxxx.h>` from the files `<yarp/os/all.h>`, `<yarp/sig/all.h>`, `<yarp/dev/all.h>`, this way the update of `yarp/xxx/all.h` would be sufficient. 

Do you think `%include <yarp/os/all.h>` would be equivalent to what you are describing? 

(General long term comment, not something that is necessary to do now): Anyhow, I think the best way to ensure regression on this is to have (even small) unit test checking this things. Matlab is a bit in a strange situation, but I think Lua ( https://github.com/robotology/yarp/tree/master/bindings/lua/tests ) and Python ( https://github.com/robotology/yarp/tree/master/bindings/python/tests ) tests are relativly easy to add.

**Comment by nunoguedelha** [comment]
Fixed by https://github.com/robotology/yarp/pull/1811.

**Comment by nunoguedelha** [comment]
> Do you think %include <yarp/os/all.h> would be equivalent to what you are describing?

Yes indeed, it should be equivalent. Isn't there a reason it wasn't done in the first place? well I can give it a quick try..
There are one or two things to check though:
- the blocks of %include are monolithic (1 block for all the `%include <yarp/dev/xxx.h>`, another for all the `%include <yarp/sig/xxx.h>`, ...), except for `%include <yarp/os/IConfig.h>` which has been moved after the call to macros `MAKE_COMMS(Property)` and `MAKE_COMMS(Bottle)`. We have to make sure moving that include back doesn't break anything
- `yarp/xxx/all.h` might have instructions that SWIG struggles processing.
Anyway, if the bindings generated after the change are the same, we're good to go.

> (General long term comment, not something that is necessary to do now): Anyhow, I think the best way to ensure regression on this is to have (even small) unit test checking this things.

Absolutely! I actually already have some unit tests in the calibration framework. We could migrate them into the **YARP** repo.

**Comment by nunoguedelha** [comment]
@traversaro , for the following list of includes in `yarp.i` we don't find the respective #include <header> in `yarp/dev/all.h`:
```
%include <yarp/os/ContactStyle.h>'
%include <yarp/os/IConfig.h>'
%include <yarp/os/NameStore.h>'  
%include <yarp/os/PortReport.h>' 
%include <yarp/os/QosStyle.h>'   
%include <yarp/os/ResourceFinder.h>'     
%include <yarp/os/TypedReader.h>'        
%include <yarp/os/TypedReaderCallback.h>'
%include <yarp/os/TypedReaderThread.h>'  
```
Is this a mistake? I guess they should be there since they are interface headers.

The other way around, `yarp/os/all.h` has a set of extra includes not present in the `yarp.i`:
```
#include <yarp/os/Clock.h>'         
#include <yarp/os/Event.h>'         
#include <yarp/os/LockGuard.h>'     
#include <yarp/os/MessageStack.h>'  
#include <yarp/os/Mutex.h>'         
#include <yarp/os/NetFloat32.h>'    
#include <yarp/os/NetFloat64.h>'    
#include <yarp/os/NetUint16.h>'     
#include <yarp/os/NetUint32.h>'     
#include <yarp/os/NetUint64.h>'     
#include <yarp/os/NetUint8.h>'      
#include <yarp/os/NetworkClock.h>'  
#include <yarp/os/Node.h>'          
#include <yarp/os/Os.h>'            
#include <yarp/os/PortInfo.h>'      
#include <yarp/os/Publisher.h>'     
#include <yarp/os/RecursiveMutex.h>'
#include <yarp/os/Subscriber.h>'    
#include <yarp/os/SystemClock.h>'   
#include <yarp/os/Terminator.h>'    
#include <yarp/os/Wire.h>'          
```

**Comment by nunoguedelha** [comment]
I don't know what is the criteria for a header to be in the `yarp/xxx/all.h` file. We can discuss further on Monday.

**Comment by drdanz** [comment]
Fixed by #1811.
